andi/andi.py

```diff
@@ -53,6 +54,52 @@ def inspect(class_or_func: Callable) -> Dict[str, List[Optional[Type]]]:
     return res
 
 
+def _inspect(class_or_func: Callable) -> List[Tuple[str, List[Optional[Type]], bool]]:
+    """
+    Return a list of tuples, 1 tuple per parameter of *class_or_func*, where
+    each tuple contains the parameter name, a list of possible types, and
+    whether or not the parameter has a default value.
+
+    .. note:: It is a modified version of :func:`inspect` to also indicate
+        whether parameters have a default value or not.
```

Comment:
Hm, why is it required to copy & modify the built-in inspect? I.e. why not have a separate function which returns if a single argument has a default value (or a function which does it for all arguments), and then combine the results?